### ISSUE TITLE
slider does not show all the thumbnails randomly

### ISSUE THREAD

**Issue by Leftyx** [sachinchoolur/lightslider#128]
I have been struggling with this issue for quite a while. 
It seems to work at some point but randomly it has the same behaviour.

My slider contains a few pictures but sometimes it shows only one of them. If I refresh the page or click on it all the others get displayed.

As you can see on the first image attached the slider contains only one image but if you look at the html it defines 5 thumbnails. The style attribute defines a width of 60px:

![lightslider_1](https://cloud.githubusercontent.com/assets/171024/7433078/b54fa932-f025-11e4-9d89-a56a059163ef.jpg)

When I refresh the page it displays all the thumbnails and the width is correct (350px):

![lightslider_2](https://cloud.githubusercontent.com/assets/171024/7433136/336aee26-f026-11e4-9383-04d1548f10d3.jpg)

```
var imageSlider = $("#dc-main-images-slider .dc-images-slider");
imageSlider.lightSlider({
    autoWidth: true,
    keyPress: true,
    thumbItem: 6,
    pager: false,
    onSliderLoad: function () {
        imageSlider.removeClass('cS-hidden');
        }
    });
```

This is the code I use to build the slider.

I've added the refresh just in case with no improvements:

```
setTimeout(function () {
    imageSlider.refresh();
    imageSlider.lightGallery();
}, 300);
```

**Comment by sachinchoolur** [comment]
Hi,
Could you please provide a jsfiddle demo or your website url is possible .

**Comment by Leftyx** [comment]
I wish I could. It's a random issue.

I've inserted some logs in lightslider code:

```
refresh.calWidth = function (cln) {
    var ln = cln === true ? $slide.find('.lslide').length : $children.length;
    console.log('children: ' + ln);
    if (settings.autoWidth === false) {
	w = ln * (slideWidth + settings.slideMargin);
    } else {
	w = 0;
	for (var i = 0; i < ln; i++) {
	    w += (parseInt($children.eq(i).width()) + settings.slideMargin);
	}
    }
    console.log('width: ' + w);
    return w;
};
```
and indeed it counts the right number of children but the width is wrong.

I guess because the thumbnails are generated by the server which crops the main image and sometimes it might take up to one second.

the only work-around I've found is to define the width of the li element in the css:

```
.dc-images-slider li {
     width: 60px;
 }
```

**Comment by sachinchoolur** [comment]
Hi,
I totally forgot about this issue. Sorry for that and hope it works fine.

**Comment by goran-despotoski** [comment]
Ok, so I had the very same issue. I found that the issue was in the refresh.calWidth when the images were not loaded yet (wrong width was calculated for some images and the total width of the container was also wrong and that is why the images were floating below). I solved it in php by getting the width of each image and adding it to the img style attribute. 

I'm sure it can be also solved by using an image preloader to make sure all the images are loaded first, but didn't have time to try and debug for my case.
